core/src/main/java/org/mskcc/cbio/portal/servlet/QueryBuilder.java

```diff
@@ -728,111 +785,42 @@ private int countProfiles (ArrayList<GeneticProfile> profileList, GeneticAlterat
         }
         return counter;
     }
-}
-
-class CohortDetails {
-
-    private Map<String, Set<String>> studySampleMap = new HashMap<>(); // <cancer study Id: set of samples>
-    private String cohortId;
-    private Boolean isVirtualStudy = false;
     
-    private Set<String> unKnownStudies = new HashSet<>();
-
-    public CohortDetails(String[] inputCohortIds, Boolean _isVirtualStudy) {
-        isVirtualStudy = _isVirtualStudy;
-        studySampleMap = filterStudySampleMap(getInputStudySampleMap(inputCohortIds));
-    }
-
-    private Map<String, Set<String>> getInputStudySampleMap(String[] inputCohortIds) {
-        Map<String, Set<String>> studySampleMap = new HashMap<>();
+    private Set<VirtualStudy> getStudies(Set<String> inputCohortIds) throws DaoException {
+        Set<VirtualStudy> studies = new HashSet<>();
         SessionServiceUtil sessionServiceUtil = new SessionServiceUtil();
         for (String inputCohortId: inputCohortIds) {
             try {
                 CancerStudy cancerStudy = DaoCancerStudy.getCancerStudyByStableId(inputCohortId);
                 // is virtual study
                 if (cancerStudy == null) {
-                		VirtualStudy virtualCohort = sessionServiceUtil.getVirtualStudyData(inputCohortId);
+                    VirtualStudy virtualCohort = sessionServiceUtil.getVirtualStudyData(inputCohortId);
```

Comment:
Can we use virtualStudy as variable name?

Reply:
sure!